### ISSUE TITLE
Followed documentation to setup CRA + Typescript and cannot discover stories

### ISSUE THREAD

**Issue by rohaanhamid** [storybookjs/storybook#8214]
I followed the documentation located at: https://storybook.js.org/docs/configurations/typescript-config/#setting-up-typescript-with-babel-loader

But no stories.tsx files are discovered from the stores folder. There are no errors on the browser console or the CLI. 

If I move stories.tsx files inside the src folder they are discovered.

**To Reproduce**
1. Create a React app with CRA with the Typescript option
2. Follow the documentation from the link above to install typescript

**Expected behavior**
stories.tsx files in the stories folder should be discovered

Please note, this is a brand new project and all the config files exactly from the snippets provided in the link above.

**Comment by kevbost** [comment]
@rohaanhamid I'm not sure if your issue is the same as mine because you mention being able to move the location of your stories and saw them appear, but I spent all morning on this and finally found this commit in the storybook PRs: https://github.com/storybookjs/storybook/pull/8186

The typescript setup docs haven't been updated to reflect some changes.

The `storiesOf` stories were loading, but no `CSF` stories.

`.storybook/config.ts`
```jsx
import { configure } from '@storybook/react';
// automatically import all files ending in *.stories.tsx
configure(require.context('../src', true, /\.stories\.tsx?$/), module)
```

After I made this change all of the missing `CSF` stories appeared.

**Comment by rohaanhamid** [comment]
@kevbost I think you are right. I switched to `storiesOf` after moving the stories and that might have caused them to appear.

**Comment by JKillian** [comment]
Hit this same issue experimenting today, where `storiesOf` worked, but CSF stories didn't work in TypeScript. I changed my `config.js` to match [@kevbost's suggestion](https://github.com/storybookjs/storybook/issues/8214#issuecomment-536694015) and it fixed the issue for me!

**Comment by kevbost** [comment]
FWIW, I discovered on accident that storybook config styles and stories can remain as javascript if allowJs. I assumed this wouldn't work, but I reverted all of my projects storybook files back to their original state and it ran as it previously did, except with typescript source files.

I discovered a different problem when converting a single storiesOf to CSF, where it brought my computer down to a crawl.

^ I think it's related but am not entirely sure yet. I'll write a new issue tomorrow.
